libr/anal/dwarf_process.c

```diff
@@ -1188,9 +1188,11 @@ static st32 parse_function_args_and_vars(Context *ctx, ut64 idx, RStrBuf *args,
 					}
 					var->type = strdup (r_strbuf_get (&type));
 					r_list_append (variables, var);
+				} else {
+					free (var->location);
+					free (var);
 				}
 				argNumber++;
```

Comment:
if the argument is ignored (thus we entered the `else` branch) shall this be incremented anyway?

Reply:
I think yes, because that just means that the argument is there, but we just wasn't able to apply that information, so I'd keep the correct number of the argument so I wouldn't misleading the user

Reply:
ok ;)